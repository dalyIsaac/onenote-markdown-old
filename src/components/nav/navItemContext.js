import React from "react";
import { Icon } from "office-ui-fabric-react/lib/Icon";
import PropTypes from 'prop-types';
import "./navItemContext.css";

export default class NavItemContext extends React.Component {
    constructor(props) {
        super(props);
        this.onClick = this.onClick.bind(this);
    }
    render() {
        return (
            <button className="navItemContextWrapper" onClick={this.onClick}>
                <div>
                    <Icon iconName={this.props.iconName} className="icon" />
                    <label>{this.props.text}</label>
                </div>
            </button>);
    }

    onClick() {
        this.props.onClick();
        if (this.props.calloutDismiss !== undefined) {
            this.props.calloutDismiss();
        }
    }
}

NavItemContext.propTypes = {
    text: PropTypes.string.isRequired,
    iconName: PropTypes.string,
    onClick: PropTypes.func.isRequired,
    calloutDismiss: PropTypes.func
}