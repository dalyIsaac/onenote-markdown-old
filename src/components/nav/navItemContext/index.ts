import NavItemContext from "./navItemContext";
export default NavItemContext;
