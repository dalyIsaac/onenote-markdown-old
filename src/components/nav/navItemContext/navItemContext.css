.navItemContextWrapper {
    background: white;
    border: none;
    width: 160px;
    text-align: left;
    cursor: pointer;
    outline: none;
    padding: 6px;
}

.navItemContextWrapper:hover {
    background: #fceffb;
}

.navItemContextWrapper:active {
    background: #f7e5f5;
}

.icon {
    vertical-align: middle;
}

label {
    cursor: pointer;
}
