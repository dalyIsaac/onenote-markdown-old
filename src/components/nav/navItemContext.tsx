import { Icon } from "office-ui-fabric-react/lib-commonjs/Icon";
import * as React from "react";
import "./navItemContext.css";

interface IPropsNavItemContext {
  text: string;
  iconName: string;
  onClick(): React.MouseEvent;
  calloutDismiss(): void;
}

export default class NavItemContext extends React.Component<
  IPropsNavItemContext
> {
  constructor(props: IPropsNavItemContext) {
    super(props);
    this.onClick = this.onClick.bind(this);
  }
  public render() {
    return (
      <button className="navItemContextWrapper" onClick={this.onClick}>
        <div>
          <Icon iconName={this.props.iconName} className="icon" />
          <label>{this.props.text}</label>
        </div>
      </button>
    );
  }

  public onClick() {
    this.props.onClick();
    if (this.props.calloutDismiss !== undefined) {
      this.props.calloutDismiss();
    }
  }
}
