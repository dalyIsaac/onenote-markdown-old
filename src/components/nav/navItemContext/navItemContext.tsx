import { Icon } from "office-ui-fabric-react";
import * as React from "react";
import "./navItemContext.css";

export interface IPropsNavItemContext {
  text: string;
  iconName: string;
  onClick(): React.MouseEvent;
  calloutDismiss(): void;
}

export default class NavItemContext extends React.Component<
  IPropsNavItemContext
> {
  constructor(props: IPropsNavItemContext) {
    super(props);
    this.onClick = this.onClick.bind(this);
  }
  public render() {
    return (
      <button className="navItemContextWrapper" onClick={this.onClick}>
        <div>
          <Icon iconName={this.props.iconName} className="icon" />
          <label>{this.props.text}</label>
        </div>
      </button>
    );
  }

  public onClick() {
    this.props.onClick();
    this.props.calloutDismiss();
  }
}
