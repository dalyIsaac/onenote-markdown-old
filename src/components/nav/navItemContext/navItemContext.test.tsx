import * as enzyme from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import { Icon, setIconOptions } from "office-ui-fabric-react";
import * as React from "react";
import * as renderer from "react-test-renderer";
import { IPropsNavItemContext } from "./navItemContext";
import NavItemContext from "./navItemContext";

setIconOptions({
  disableWarnings: true
});

enzyme.configure({ adapter: new Adapter() });

function setUp() {
  const props: IPropsNavItemContext = {
    calloutDismiss: jest.fn(),
    iconName: "ChromeClose",
    onClick: jest.fn(),
    text: "Hello world"
  };
  const wrapper = enzyme.mount(<NavItemContext {...props} />);
  return { props, wrapper };
}

describe("Components: NavItemContext", () => {
  test("It should render self and subcomponents", () => {
    const { props, wrapper } = setUp();

    const button = wrapper.find("button").first();
    expect(button.hasClass("navItemContextWrapper")).toBe(true);

    const div = wrapper.find("div").first();
    expect(div.parent().is("button.navItemContextWrapper")).toBe(true);

    const icon = wrapper.find(Icon).first();
    expect(icon.hasClass("icon")).toBe(true);
    expect(icon.props().iconName).toBe(props.iconName);
    expect(icon.parent().is("div")).toBe(true);

    const label = wrapper.find("label").first();
    expect(label.text()).toBe(props.text);
    expect(label.parent().is("div")).toBe(true);

    expect(
      renderer.create(<NavItemContext {...props} />).toJSON()
    ).toMatchSnapshot();
  });

  test("Ensures that onClick fires the correct methods", () => {
    const { wrapper } = setUp();
    const navItemContext = wrapper.instance() as NavItemContext;
    navItemContext.onClick();
    expect((navItemContext.props.onClick as jest.Mock).mock.calls.length).toBe(
      1
    );
    expect(
      (navItemContext.props.calloutDismiss as jest.Mock).mock.calls.length
    ).toBe(1);
  });
});
