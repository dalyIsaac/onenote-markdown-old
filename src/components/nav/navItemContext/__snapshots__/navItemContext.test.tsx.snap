// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: NavItemContext It should render self and subcomponents 1`] = `
<button
  className="navItemContextWrapper"
  onClick={[Function]}
>
  <div>
    <i
      aria-hidden={true}
      className="icon root-42"
      data-icon-name="ChromeClose"
      role="presentation"
    />
    <label>
      Hello world
    </label>
  </div>
</button>
`;
